docs/OpenShift GitOps CLI User Guide.md

```diff
@@ -0,0 +1,636 @@
+# OpenShift GitOps CLI User Guide
+OpenShift GitOps `argocd` CLI enables users to manage OpenShift GitOps from a terminal. Using the CLI tool, users can manage ArgoCD resources (Application, ApplicationSets, AppProjects, User accounts, GPG keys etc) from a client terminal.
+
+**Note:** Both the archives and the RPMs contain the `argocd` executable binary file.
+
+**WARNING** 	
+Red Hat OpenShift GitOps with the `argocd` CLI tool is a Technology Preview feature only.
+
+**Tip:** This guide assumes you have a grounding in the tools that OpenShift GitOps is based on. Please read [about Red Hat OpenShift GitOps](https://docs.openshift.com/gitops/1.11/understanding_openshift_gitops/about-redhat-openshift-gitops.html) to learn about these tools.
+
+## Installing OpenShift GitOps CLI (argocd)
+Use the CLI tool to manage Red Hat OpenShift GitOps from a terminal. You can install the CLI tool on different platforms.
+
+### Installing the Red Hat OpenShift GitOps CLI on Linux using an RPM
+For Red Hat Enterprise Linux (RHEL) version 8 or later, you can install the Red Hat OpenShift GitOps CLI as an RPM.
+#### Prerequisites
+You have an active OpenShift Container Platform subscription on your Red Hat account.
+You have root or sudo privileges on your local system.
+#### Procedure
+- Register with Red Hat Subscription Manager:
+```
+# subscription-manager register
+```
+- Pull the latest subscription data:
+```
+# subscription-manager refresh
+```
+- List the available subscriptions:
+```
+# subscription-manager list --available --matches '*gitops*'
+```
+- In the output for the previous command, find the pool ID for your OpenShift Container Platform subscription and attach the subscription to the registered system:
+```
+# subscription-manager attach --pool=<pool_id>
+```
+- Enable the repositories required by Red Hat OpenShift GitOps:
+    - Linux (x86_64, amd64)
+        ```
+        # subscription-manager repos --enable="gitops-1.12-for-rhel-8-x86_64-rpms"
+        ```
+    - Linux on IBM zSystems and IBM® LinuxONE (s390x)
+        ```
+        # subscription-manager repos --enable="gitops-1.12-for-rhel-8-s390x-rpms"
+        ```
+    - Linux on IBM Power (ppc64le)
+        ```
+        # subscription-manager repos --enable="gitops-1.12-for-rhel-8-ppc64le-rpms"
+        ```
+    - Linux on ARM (aarch64, arm64)
+        ```
+        # subscription-manager repos --enable="gitops-1.12-for-rhel-8-aarch64-rpms"
+        ```
```

Comment:
@anandf FOr RHEL 9, would this the repo inputs be the following ones?

- "gitops-1.12-for-rhel-9-x86_64-rpms"
- "gitops-1.12-for-rhel-9-s390x-rpms"
- "gitops-1.12-for-rhel-9-ppc64le-rpms"
- "gitops-1.12-for-rhel-9-aarch64-rpms"

Reply:
@anandf Please provide clarifications for these queries: 

- For ROSA, OSD, and ARO distros, will all these steps, command, and procedures hold good as is?
- Is there any difference between what GitOps offers OCP v/s what we do for OSD, ROSA, and ARO distros?
- Can you please confirm that there are no changes or special configurations needed to use GitOps `argocd` CLI on OSD, ROSA,  and ARO distros out of the box?

Thank you!